src/include/homestore/btree/mem_btree.hpp

```diff
@@ -43,7 +43,7 @@ class MemBtree : public Btree< K, V > {
     btree_status_t write_node_impl(const BtreeNodePtr& node, void* context) { return btree_status_t::success; }
 
     btree_status_t read_node_impl(bnodeid_t id, BtreeNodePtr& node) const override {
-        node = BtreeNodePtr{r_cast< BtreeNode* >(id)};
+        node.reset(r_cast< BtreeNode* >(id));
```

Comment:
It's not clear to me what issue this change fixed?

Reply:
the way that that has been implemented it causes the reference count for the intrusive pointer to bump up. As a result, in [post_order_traversal](https://raw.githubusercontent.com/eBay/HomeStore/fiber/src/include/homestore/btree/detail/btree_common.ipp#:~:text=read_and_lock_node(m_root_node_info.bnode_id()%2C%20root%2C%20ltype%2C%20ltype%2C%20nullptr)%3B), BtreeNodePtr root has not release the pointer.

Reply:
Thanks for the explanation!